### DISCUSSION TITLE
KIE中的RE任务，运行官方示例没有linking信息

### DISCUSSION THREAD

**Discussion by cwmore** [PaddlePaddle/PaddleOCR#13840] [Q&A] [answered]
我搞了两天了，没定位到问题，是paddleocr的代码问题还是官方给的模型问题？：

我使用如下命令做的推理，用的也是官方给的推理模型，使用的是XFUND数据集，图片也是官方给的：
python3 kie/predict_kie_token_ser_re.py \
  --kie_algorithm=LayoutXLM \
  --re_model_dir=../inference/re_vi_layoutxlm_xfund_infer \
  --ser_model_dir=../inference/ser_vi_layoutxlm_xfund_infer \
  --use_visual_backbone=False \
  --image_dir=./docs/kie/input/zh_val_42.jpg \
  --ser_dict_path=../train_data/XFUND/class_list_xfun.txt \
  --vis_font_path=../doc/fonts/simfang.ttf \
  --ocr_order_method="tb-yx"

运行完后的结果图片上是有连接线关系的，
![截图_选择区域_20240909100941](https://github.com/user-attachments/assets/9172077d-65f0-4a6b-b755-a1ac1f40fbe8)

**但是./output/infer.txt中没有linking信息，找不到QUESTION和ANSWER的关系，**
![截图_选择区域_20240909100723](https://github.com/user-attachments/assets/1f6c9f90-d124-46fe-abdb-a89e07ffdc12)

求大佬指点？

运行预训练的模型，并带上标注好的验证集的数据（含linking信息）,运行后是会有linking信息的，但是我都手动标注了，还要模型去推理干什么呢，

类似问题：https://github.com/PaddlePaddle/PaddleOCR/discussions/12241

我本地的依赖如下：
![image](https://github.com/user-attachments/assets/cd55db49-cd31-43a1-a136-68cff0a60aa9)

**Comment by GreatV** [comment]
老哥你多看几眼代码就知道了

https://github.com/PaddlePaddle/PaddleOCR/blob/5b54ac460623d7dd23717e39b43b182a359bd606/tools/infer_kie_token_ser_re.py#L219

https://github.com/PaddlePaddle/PaddleOCR/blob/5b54ac460623d7dd23717e39b43b182a359bd606/ppocr/utils/visual.py#L88-L132

**Comment by cwmore** [comment, marked as answer]
感谢大佬答疑，我也debug了一下代码，确实发现result中是以元组顺序来存的QA：
![image](https://github.com/user-attachments/assets/febb733a-5281-4602-a9ff-0832b9994461)
所以之前用标注linking跑出来的results中的linking应该只是用于训练的吧？@GreatV 
![image](https://github.com/user-attachments/assets/7ceace7d-199a-47a6-95c6-943bf1798e24)

被之前的文档引导偏了，另外，官方的这个RE示例不应该直接输出用户最关心的关键信息吗，比如：
{
"姓名":"张三",
xxx
}
